src/Abstractions/TaskFailureDetails.cs

```diff
@@ -156,6 +157,15 @@ internal CoreFailureDetails ToCoreFailureDetails()
 
         if (exception is CoreOrchestrationException coreEx)
         {
+            if (coreEx is CoreSubOrchestrationFailedException)
+            {
+                return new TaskFailureDetails(
+                    coreEx.GetType().ToString(),
+                    coreEx.Message,
+                    coreEx.StackTrace,
+                    FromCoreFailureDetailsRecursive(coreEx.FailureDetails?.InnerFailure) ?? FromExceptionRecursive(coreEx.InnerException));
```

Comment:
Should this be
FromCoreFailureDetailsRecursive(coreEx.FailureDetails) ?? FromExceptionRecursive(coreEx.InnerException))
To prevent loss of the outer FailureDetails?